src/lib/utils/resource-loader.ts

```diff
@@ -275,3 +279,57 @@ export const loadFormatter = (formatterId: string): IFormatter => {
 export const loadParser = (parserId: string): Parser => {
     return loadResource(parserId, TYPE.parser);
 };
+
+const searchNpmPackages = (searchTerm: string): Promise<Array<NpmPackage>> => {
+    return new Promise((resolve, reject) => {
+        const results = [];
+
+        const searchOptions = {
+            description: true,
+            excluded: [],
+            include: [searchTerm],
+            limit: 1000,
+            staleness: 900,
+            unicode: false
+        };
+
+        esearch(searchOptions)
+            .on('data', (data) => {
+                results.push(data);
+            })
+            .on('error', (err) => {
+                reject(err);
+            })
+            .on('end', () => {
+                resolve(results);
+            });
+    });
+};
+
+const loadNpm = () => {
+    return npmLoadAsync({ loaded: false });
+};
+
+export const getNpmPackages = async (searchTerm: string = 'sonarwhal'): Promise<Array<NpmPackage>> => {
+    await loadNpm();
+
+    return searchNpmPackages(searchTerm);
+};
+
+const filterPackages = (packages: Array<NpmPackage>, initTerm: string) => {
+    return packages.filter((pkg) => {
+        return pkg.name.startsWith(initTerm);
+    });
+};
+
+export const getExternalRulesFromNpm = async () => {
+    const rules = await getNpmPackages('sonarwhal-rule');
+
+    return filterPackages(rules, 'sonarwhal-rule');
+};
+
+export const getCoreRulesFromNpm = async () => {
+    const rules = await getNpmPackages('@sonarwhal/rule');
+
+    return filterPackages(rules, '@sonarwhal/rule');
```

Comment:
Can you put a comment explaining why we are filtering even though you are searching for `@sonarwhal/rule`?

Maybe we could also do a search just by `@sonarwhal`  and then do the grouping in memory by rule/parser/connector/etc.?

Reply:
I don't know if it is because I'm doing something weird or not, but the search doesn't work like you expect to work. With the packages there is right now, search for `@sonarwhal` or `sonarwhal` return the same elements (and not, not all the packages start with @sonarwhal`.

About the filter, I will write some explanation, but look for `@sonarwhal/rule` doesn't have to return only packages that start with `@sonarwhal/rule` :(